### PR TITLE
external packages: use CPM packages instead of third-party directory

### PR DESCRIPTION
**Pull request by arogge**
This is an alternative PR to #1911

* write a cpm-packages.yaml with a bit of metadata to `$CMAKE_BINARY_DIR`
* add a `CPM_ONLY`-mode to our CMake, so you can run only CPM
* extend update_license.py to grab the licenses of the CPM packages
* replace (and update) all third-party dependencies with CPM packages
* update all pip-tools
* fix some minor issues (fastlz, truncate systemtest)

TODO:
* [x] make `pr-tool update-license` work with CPM licenses
* [x] clean up duplicates in `devtools/template/LICENSE.txt` (some CPM licenses already exist, so we'd have duplicates)

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] ~Required backport PRs have been created~
- [x] Correct milestone is set

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

##### Tests
- [x] Decision taken that a test is required (if not, then remove this paragraph)
- [x] The choice of the type of test (unit test or systemtest) is reasonable
- [x] Testname matches exactly what is being tested
- [x] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -240,11 +238,11 @@ Package:        bareos-storage-dedupable
 Architecture:   any
 Pre-Depends:    debconf (>= 1.4.30) | debconf-2.0
 Depends:        bareos-common (= ${binary:Version}), bareos-storage (= ${binary:Version}), lsb-base (>= 3.2-13), ${shlibs:Depends}, ${misc:Depends}
-Description: Backup Archiving Recovery Open Sourced - storage daemon DEDUPABLE backend
+Description: Backup Archiving Recovery Open Sourced - storage daemon DEDUP backend
```
Comment (pstorz) [inline debian/control:241]:
```suggestion
Description: Backup Archiving Recovery Open Sourced - storage daemon DEDUPABLE backend
```

File: debian/control [hunk 2]
```diff
@@ -240,11 +238,11 @@ Package:        bareos-storage-dedupable
...
  Bareos is a set of programs to manage backup, recovery and verification of
  data across a network of computers of different kinds.
  .
- This package contains the Storage backend for the dedupable storage format.
+ This package contains the Storage backend for DEDUP files.
```
Comment (pstorz) [inline debian/control:245]:
```suggestion
 This package contains the Storage backend for DEDUPABLE files.
```

**Review by pstorz — APPROVED** [review]
Looks good to me

**Review by pstorz — CHANGES_REQUESTED** [review]
See comments

File: debian/control (hunk 1 above)
Comment (arogge) [reply debian/control:241]: fixed in #1977 and #1978

File: debian/control (hunk 2 above)
Comment (arogge) [reply debian/control:245]: fixed in #1977 and #1978

**Comment by arogge** [comment]
The changes you requested were handled in PR #1977 and #1978.

**Review by pstorz — APPROVED** [review]

**Pull request merged**